### PR TITLE
Fix reuse of ProjectDescriptionHelpers cache

### PR DESCRIPTION
**Pull request by danibachar**
### Short description 📝

This PR aiming to fix the following issue.
Using the dump (or basically any tuist command) can trigger the build of the ProjectDescriptionHelpers module.
Tuist is hashing and creating a path to that module using these two methods [1](https://github.com/tuist/tuist/blob/main/Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersHasher.swift#L35), [2](https://github.com/tuist/tuist/blob/main/Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersHasher.swift#L55) (one for the actual folder path, and the second for the parent folder path).
I'm not sure why Tuist would like to delete the parent folder and as such potentially delete other cached builds of similar modules.

### How to test the changes locally 🧐

Here are example steps that will cause Tuist to rebuild the ProjectDescriptionHelpers, which could have been cached before

1) `tuist dump --verbose --path pathToTuist/fixtures/app_with_spm_dependencies/app_with_spm_dependencies/Features/FeatureOne/`
2) `TUIST_SOME_ENV_VAR tuist dump --verbose --path pathToTuist/fixtures/app_with_spm_dependencies/app_with_spm_dependencies/Features/FeatureOne/`
3) `tuist dump --verbose --path pathToTuist/fixtures/app_with_spm_dependencies/app_with_spm_dependencies/Features/FeatureOne/`
4) `TUIST_SOME_ENV_VAR tuist dump --verbose --path pathToTuist/fixtures/app_with_spm_dependencies/app_with_spm_dependencies/Features/FeatureOne/`

Note that the ProjectDescriptionHelpers module will be built every-time because we have deleted the parent folder including the previously cached module.

Explanation:
Adding an env var will cause a miss in the code above checking for existing cached ProjectDescriptionHelpers module. This makes sense as Tuist takes Env vars into account when [hashing](https://github.com/tuist/tuist/blob/main/Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersHasher.swift#L41).

The deletion of the whole parent folder just causing loosing of the previously built cache.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift [hunk 1]
```diff
@@ -176,12 +176,6 @@ public final class ProjectDescriptionHelpersBuilder: ProjectDescriptionHelpersBu
             return projectDescriptionHelpersModule
         }
 
-        // If the same helpers directory has been previously compiled
-        // we delete it before compiling the new changes.
-        if FileHandler.shared.exists(helpersCachePath) {
-            try await fileSystem.remove(helpersCachePath)
-        }
```
Comment (fortmarek) [inline Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift:183]: Should we still delete the helpers directory but instead delete only the `helpersModuleCachePath` directory?

**Review by fortmarek — COMMENTED** [review]
I don't see why we'd need to delete the `helpersCachePath` instead of the individual of the `helpersModuleCachePath`.

@kwridan @pepicrft anyone has an idea why we're currently removing `helpersCachePath` and if there is a good reason for that?

File: Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift (hunk 1 above)
Comment (danibachar) [reply Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift:183]: How would you know that path? It is based on a hash we dont posses for this run.

Comment (danibachar) [reply Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift:183]: Not that the helpersModuleCachePath is guaranteed to not be present (checking it in the line above)

Comment (danibachar) [reply Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift:183]: Im guessing you want to clear other cached builds? If so how would you distinguish those? You don’t have the other hashes. Im still not convinced we should delete any cache here. Let the user decide when to empty it with tuist clean

Comment (fortmarek) [reply Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift:183]: ahh, you are right, my bad. Then I'm fully onboard with this change 👍

**Comment by fortmarek** [comment]
I wonder if this could also help fix [this issue](https://github.com/tuist/tuist/issues/6374) raised by @Ernest0-Production

**Comment by danibachar** [comment]
About the acceptance tests failure - they fail also on main for me...

**Comment by fortmarek** [comment]
> About the acceptance tests failure - they fail also on main for me...

Fixed: https://github.com/tuist/tuist/pull/6546

Merging latest changes from `main` should resolve that for you 🙂

**Comment by pepicrft** [comment]
> @kwridan @pepicrft anyone has an idea why we're currently removing helpersCachePath and if there is a good reason for that?

Not that I can think of. If acceptance tests pass, we should be confident about the change.

@danibachar once you rebase and we get CI passing we can merge.

**Review by pepicrft — APPROVED** [review]

**Review by fortmarek — APPROVED** [review]
Would you mind rerunning the pipeline? It might be the failing acceptance test is flaky as it works on my machine 🤔

**Comment by danibachar** [comment]
> Would you mind rerunning the pipeline? It might be the failing acceptance test is flaky as it works on my machine 🤔

BTW - For some reason clicking the retry button doesn't really work, it just says "You have successfully requested Tuist generator acceptance tests be rerun." - might be a permissions issue?

I had to push a dummy commit to trigger tests...

**Comment by fortmarek** [comment]
> BTW - For some reason clicking the retry button doesn't really work, it just says "You have successfully requested Tuist generator acceptance tests be rerun." - might be a permissions issue?

We'll take a look, looks like the Codemagic setup needs some work for the forks.

**Comment by Ernest0-Production** [comment]
> I wonder if this could also help fix [this issue](https://github.com/tuist/tuist/issues/6374) raised by @Ernest0-Production

Hi! Thanks for mentioning me. In what release version is this fix expected so that I can test it on a working project?

**Comment by fortmarek** [comment]
This is in the latest release, 4.22.0

**Comment by Ernest0-Production** [comment]
Thanks!

@pepicrft It seems you forgot to indicate this in the release Changelog)

**Pull request merged**